harness/determined/_trial_controller.py

```diff
@@ -279,27 +279,32 @@ def __init__(self, *args: List[Any], **kwargs: Dict[str, Any]) -> None:
         )
 
         # Initialize communication directly between training processes.
-        self.train_process_comm_chief = None  # type: Optional[ipc.ZMQServer]
-        self.train_process_comm_worker = None  # type: Optional[ipc.ZMQClient]
+        self.train_process_comm_chief = None  # type: Optional[ipc.ZMQBroadcastServer]
+        self.train_process_comm_worker = None  # type: Optional[ipc.ZMQBroadcastClient]
         if self.hvd_config.use:
             self._initialize_train_process_comm()
 
     def _initialize_train_process_comm(self) -> None:
         check.true(self.hvd_config.use)
+
+        srv_pub_port = constants.INTER_TRAIN_PROCESS_COMM_PORT_1
+        srv_pull_port = constants.INTER_TRAIN_PROCESS_COMM_PORT_2
+
         if self.is_chief:
-            logging.debug(
-                f"Chief {hvd.rank()} setting up server with "
-                f"port {constants.INTER_TRAIN_PROCESS_COMM_PORT}."
-            )
-            self.train_process_comm_chief = ipc.ZMQServer(
-                ports=[constants.INTER_TRAIN_PROCESS_COMM_PORT], num_connections=1
+            logging.debug("Chief setting up server with ports " f"{srv_pub_port}/{srv_pull_port}.")
```

Comment:
non-blocking: this sting format looks strange.

Reply:
oops, `black` autoformatting got me.  Fixed.